appservice/src/intTest/groovy/nu/yona/server/UserTest.groovy

```diff
@@ -1,236 +1,227 @@
+			testUser(john, false, ts)
 
 		cleanup:
-		if (userURL) {
-			appService.deleteUser(userURL, password)
-		}
+			appService.deleteUser(johnAsCreated)
 	}
 
-	def 'Update John Doe with a different mobile number'(){
+	def 'Update John Doe with the same mobile number'()
+	{
 		given:
-		def userAddResponse = appService.addUser(userCreationJSON, password);
-		def userURL = appService.stripQueryString(userAddResponse.responseData._links.self.href);
-		def confirmationCode = userAddResponse.responseData.confirmationCode;
-		def confirmResponse = appService.confirmMobileNumber(userURL, """ { "code":"${confirmationCode}" } """, password)
-		def newMobileNumber = "+${timestamp}1"
+			def ts = timestamp
+			def john = createJohnDoe(ts)
+			appService.confirmMobileNumber(appService.&assertResponseStatusSuccess, john)
 
 		when:
-		def userUpdateResponse = appService.updateUser(userURL, userCreationJSON.replace("+${timestamp}", newMobileNumber), password);
-		def newConfirmationCode
-		def newConfirmResponse
-		if(userUpdateResponse.status == 200) {
-			newConfirmationCode = userUpdateResponse.responseData.confirmationCode;
-			newConfirmResponse = appService.confirmMobileNumber(userURL, """ { "code":"${newConfirmationCode}" } """, password)
-		}
+			def newNickname = "Johnny"
+			def updatedJohn = john.convertToJSON()
+			updatedJohn.nickname = newNickname
+			def userUpdateResponse = appService.updateUser(john.url, updatedJohn, john.password);
 
 		then:
-		userUpdateResponse.status == 200
-		userUpdateResponse.responseData.mobileNumberConfirmed == false
-		userUpdateResponse.responseData.mobileNumber == newMobileNumber
-		newConfirmationCode != confirmationCode
-		newConfirmResponse.status == 200
-		newConfirmResponse.responseData.mobileNumberConfirmed == true
+			userUpdateResponse.status == 200
+			userUpdateResponse.responseData._links?.confirmMobileNumber?.href == null
+			userUpdateResponse.responseData.nickname == newNickname
 
 		cleanup:
-		if (userURL) {
-			appService.deleteUser(userURL, password)
-		}
+			appService.deleteUser(john)
 	}
 
-	def 'Delete John Doe'(){
+	def 'Update John Doe with a different mobile number'()
+	{
 		given:
-		def userURL = appService.stripQueryString(appService.addUser(userCreationJSON, password).responseData._links.self.href);
+			def ts = timestamp
+			def john = createJohnDoe(ts)
+			appService.confirmMobileNumber(appService.&assertResponseStatusSuccess, john)
 
 		when:
-		def response = appService.deleteUser(userURL, password)
+			String newMobileNumber = "${john.mobileNumber}1"
+			def updatedJohn = john.convertToJSON()
+			updatedJohn.mobileNumber = newMobileNumber
+			def userUpdateResponse = appService.updateUser(john.url, updatedJohn, john.password);
 
 		then:
-		response.status == 200
-		verifyUserDoesNotExist(userURL)
+			userUpdateResponse.status == 200
+			userUpdateResponse.responseData._links?.confirmMobileNumber?.href != null
+			userUpdateResponse.responseData.mobileNumber == newMobileNumber
+
```

Comment:
I am missing the rest of the test here. In the original the new mobile number was confirmed afterwards and checked.